File: algo_trader/lib/trade.py
class Order:
    def __init__(self, price, timestamp):
        self.price: float = price
        self.timestamp: int = timestamp

class Trade:
    def __init__(self, symbol: str, amount: float, price: float, timestamp: int):
        self.symbol: str = symbol
        self.amount: float = amount
        print(f"[Trade] amount: {amount}")
        self.buy_order = Order(price, timestamp)
        self.sell_order = None
    
    def sell(self, price, timestamp):
        self.sell_order = Order(price, timestamp)

    def is_closed(self) -> bool:
        return self.sell_order is not None

File: algo_api/internal/database/trade.go
package database

import "github.com/leesper/couchdb-golang"

type Trade struct {
	PvtType string `json:"pvt_type"`
	TradePublicFields
	couchdb.Document
}

type TradePublicSell struct {
	Price     string `json:"price,omitempty"`
	Timestamp int64  `json:"timestamp,omitempty"`
}

type TradePublicFields struct {
	Pair   string `json:"pair"`
	Amount string `json:"amount"`
	Orders struct {
		Buy struct {
			Price     string `json:"price"`
			Timestamp int64  `json:"timestamp"`
		} `json:"buy"`
		Sell TradePublicSell `json:"sell,omitempty"`
	} `json:"orders"`
	StrategyID string `json:"strategy_id"`
}

type TradeResponseFields struct {
	TradePublicFields
	ID string `json:"id"`
}
